### PR TITLE
Move definition of VKB_DEBUG from header (error.h) to cmake (global_options.cmake)

### PR DESCRIPTION
**Pull request by asuessenbach**
## Description

Additionally remove all the now unnecessary includes of error.h
Compiles and runs fine with MSVC.

## General Checklist:

Please ensure the following points are checked:

- [x] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
- [x] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
- [x] I have commented any added functions (in line with Doxygen)
- [x] I have commented any code that could be hard to understand
- [x] My changes do not add any new compiler warnings
- [x] My changes do not add any new validation layer errors or warnings
- [x] I have used existing framework/helper functions where possible
- [ ] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [x] My changes do not add any regressions
- [x] I have tested every sample to ensure everything runs correctly
- [x] This PR describes the scope and expected impact of the changes I am making

### CODE REVIEW

File: bldsys/cmake/global_options.cmake [hunk 1]
```diff
@@ -52,3 +52,5 @@ add_definitions(-DROOT_PATH_SIZE=${ROOT_PATH_SIZE})
 
 set(CMAKE_C_FLAGS_DEBUG   "-DDEBUG=0 ${CMAKE_C_FLAGS_DEBUG}")
 set(CMAKE_CXX_FLAGS_DEBUG "-DDEBUG=0 ${CMAKE_CXX_FLAGS_DEBUG}")
+
+add_compile_definitions($<$<CONFIG:DEBUG>:VKB_DEBUG>)
```
Comment (TomAtkinsonArm) [inline bldsys/cmake/global_options.cmake]: We should add compile definitions per target `target_compile_definitions`

https://cmake.org/cmake/help/latest/command/target_compile_definitions.html

Comment (asuessenbach) [reply bldsys/cmake/global_options.cmake]: Why should we do that?
VKB_DEBUG probably is a some global setting, that should not be defined per target.

**Comment by gpx1000** [comment]
Because if it's defined by target and public than all targets that inherit
get it too.  So it still goes everywhere we want it.  It makes things
cleaner when we don't want/need it on 3rd party libraries that might have
something defined with it for whatever reason that modifies behavior.
It's named with vkb at the front to protect from that eventuality, but if
it's not even in the define list when compiler is working on the library
then it's not even a chance of happening.

**Comment by asuessenbach** [comment]
Ok, sounds reasonable... will move that define into sample_helper.cmake and make it per target then.

**Review by gary-sweet — APPROVED** [review]

**Review by TomAtkinsonArm — APPROVED** [review]

**Pull request merged**